src/httpbeast.nim

```diff
@@ -544,6 +560,16 @@ proc run*(onRequest: OnRequest, settings: Settings) =
         createThread[(OnRequest, Settings, bool)](
           t, eventLoop, (onRequest, settings, false)
         )
+      addExitProc(proc() =
+        for thr in threads:
+          when compiles(pthread_cancel(thr.sys)):
+            discard pthread_cancel(thr.sys)
+          if not isNil(thr.core):
+            when defined(gcDestructors):
+              c_free(thr.core)
+            else:
+              deallocShared(thr.core)
+      )
```

Comment:
What is the purpose of this?

Reply:
for fix error `SIGSEGV: Illegal storage access. (Attempt to read from nil?)`
https://github.com/dom96/httpbeast/actions/runs/5778815517/job/15660432786#step:4:34